infrastructure/zk/src/database.ts

```diff
@@ -1,101 +1,199 @@
+    }
+    if (opts.server) {
+        dals.set('core/lib/dal', process.env.DATABASE_URL!);
+    }
+    return dals;
+}
+
+function getTestDals(opts: any): Map<string, string> {
+    let dals = new Map<string, string>();
+    if (!opts.prover && !opts.server) {
+        dals.set('core/lib/dal', process.env.DATABASE_TEST_URL!);
+        dals.set('prover/prover_dal', process.env.DATABASE_PROVER_URL!);
+    }
+    if (opts.prover) {
+        dals.set('prover/prover_dal', process.env.DATABASE_PROVER_TEST_URL!);
+    }
+    if (opts.server) {
+        dals.set('core/lib/dal', process.env.DATABASE_TEST_URL!);
+    }
+    return dals;
+}
+
+async function resetTestDal(dalPath: string, dalDb: string) {
     console.log('recreating postgres container for unit tests');
     await utils.spawn('docker compose -f docker-compose-unit-tests.yml down');
     await utils.spawn('docker compose -f docker-compose-unit-tests.yml up -d');
     await wait(100);
     console.log('setting up a database template');
-    await setup();
+    await setupForDal(dalPath, dalDb);
     console.log('disallowing connections to the template');
     await utils.spawn(
-        `psql "${process.env.DATABASE_URL}" -c "update pg_database set datallowconn = false where datname = current_database()"`
+        `psql "${dalDb}" -c "update pg_database set datallowconn = false where datname = current_database()"`
     );
 }
 
-export async function drop() {
+export async function resetTest(opts: any) {
     await utils.confirmAction();
-    console.log('Dropping DB...');
-    await utils.spawn('cargo sqlx database drop -y');
+    let dals = getTestDals(opts);
+    for (const [dalPath, dalDb] of dals.entries()) {
+        await resetTestDal(dalPath, dalDb);
+    }
 }
 
-export async function migrate() {
+async function dropForDal(dalPath: string, dalDb: string) {
+    console.log(`Dropping DB for dal ${dalPath}...`);
+    await utils.spawn(`cargo sqlx database drop -y --database-url ${dalDb}`);
+}
+
+export async function drop(opts: any) {
     await utils.confirmAction();
-    console.log('Running migrations...');
-    await utils.spawn('cd core/lib/dal && cargo sqlx database create && cargo sqlx migrate run');
+    let dals = getDals(opts);
+    for (const [dalPath, dalDb] of dals.entries()) {
+        await dropForDal(dalPath, dalDb);
+    }
+}
+
+async function migrateForDal(dalPath: string, dalDb: string) {
+    console.log(`Running migrations for ${dalPath}...`);
+    await utils.spawn(
+        `cd ${dalPath} && cargo sqlx database create --database-url ${dalDb} && cargo sqlx migrate run --database-url ${dalDb}`
+    );
 }
 
-export async function generateMigration(name: String) {
+export async function migrate(opts: any) {
+    await utils.confirmAction();
+    let dals = getDals(opts);
+    for (const [dalPath, dalDb] of dals.entries()) {
+        await migrateForDal(dalPath, dalDb);
+    }
+}
+
+async function generateMigrationForDal(dalPath: string, dalDb: string, name: String) {
```

Comment:
we could have made used of typescript's strong typing here. Using enum for `dalPath`, not using `any` for opts etc. 
I'm not insisting though - if it's working like it does so we can keep as is to save time